go/vt/sqlparser/ast_permissions.go

```diff
@@ -388,6 +672,264 @@ func (d *DropRole) walkSubtree(visit Visit) error {
 	return nil
 }
 
+// GrantPrivilege represents the GRANT...ON...TO statement.
+type GrantPrivilege struct {
+	Privileges []Privilege
+	ObjectType GrantObjectType
+	PrivilegeLevel PrivilegeLevel
+	To []AccountName
+	WithGrantOption bool
+	As *GrantUserAssumption
+}
+
+var _ Statement = (*GrantPrivilege)(nil)
+
+// iStatement implements the interface Statement.
+func (g *GrantPrivilege) iStatement() {}
+
+// Format implements the interface Statement.
+func (g *GrantPrivilege) Format(buf *TrackedBuffer) {
+	buf.Myprintf("grant")
+	for i, privilege := range g.Privileges {
+		if i > 0 {
+			buf.Myprintf(",")
+		}
+		buf.Myprintf(" %s", privilege.String())
+	}
+	buf.Myprintf(" on")
+	switch g.ObjectType {
+	case GrantObjectType_Any:
+		// Do nothing, here to explicitly list all cases
+	case GrantObjectType_Table:
+		buf.Myprintf(" table")
+	case GrantObjectType_Function:
+		buf.Myprintf(" function")
+	case GrantObjectType_Procedure:
+		buf.Myprintf(" procedure")
+	}
+	buf.Myprintf(" %s to", g.PrivilegeLevel.String())
+	for i, user := range g.To {
+		if i > 0 {
+			buf.Myprintf(",")
+		}
+		buf.Myprintf(" %s", user.String())
+	}
+	if g.WithGrantOption {
+		buf.Myprintf(" with grant option")
+	}
+	if g.As != nil {
+		buf.Myprintf(" %s", g.As.String())
+	}
+}
+
+// walkSubtree implements the interface Statement.
+func (g *GrantPrivilege) walkSubtree(visit Visit) error {
+	return nil
+}
+
+// GrantRole represents the GRANT...TO statement.
+type GrantRole struct {
+	Roles []AccountName
+	To []AccountName
+	WithAdminOption bool
+}
+
+var _ Statement = (*GrantRole)(nil)
+
+// iStatement implements the interface Statement.
+func (g *GrantRole) iStatement() {}
+
+// Format implements the interface Statement.
+func (g *GrantRole) Format(buf *TrackedBuffer) {
+	buf.Myprintf("grant")
+	for i, role := range g.Roles {
+		if i > 0 {
+			buf.Myprintf(",")
+		}
+		buf.Myprintf(" %s", role.String())
+	}
+	buf.Myprintf(" to")
+	for i, user := range g.To {
+		if i > 0 {
+			buf.Myprintf(",")
+		}
+		buf.Myprintf(" %s", user.String())
+	}
+	if g.WithAdminOption {
+		buf.Myprintf(" with admin option")
+	}
+}
+
+// walkSubtree implements the interface Statement.
+func (g *GrantRole) walkSubtree(visit Visit) error {
+	return nil
```

Comment:
We don't actually use this anywhere, want to remove it from the interface?

Reply:
Yep! I've always wanted to delete this lol